core/src/ipc.rs

```diff
@@ -131,3 +135,99 @@ impl<T> FromRawFd for IpcReceiver<T> {
         }
     }
 }
+
+/// Creates a pair of sockets that are ment for passing file descriptors to IO resources of type T.
+pub fn io_pair<T>() -> TypedResult<(IoSender<T>, IoReceiver<T>)>
+where
+    T: AsRawFd,
+{
+    let (tx, rx) = socketpair(
```

Comment:
From a semantic point of view, the terms `Sender` and `Receiver` are confusing while using a unix domain socket. The sender/receiver semantic gives more the vibes of a pipe.

Reply:
They are named this way, because it is only possible to receive the sockets inside the partition and the hyperevisor will only send to it. The same semantics are used for the Socket-Pair that is used for making RPC calls.